generate_line_box.py

```diff
@@ -28,17 +28,18 @@
 # load gt
 with io.open(args.txt, "r", encoding='utf-8') as f:
     lines = f.read().strip().split('\n')
-
-for line in lines:
-    line = unicodedata.normalize('NFC', line.strip())
-    if line:
-        for i in range(1, len(line)):
-            char = line[i]
-            prev_char = line[i-1]
-            if unicodedata.combining(char):
-                print("%s 0 0 %d %d 0" % ((prev_char + char), width, height))
-            elif not unicodedata.combining(prev_char):
-                print("%s 0 0 %d %d 0" % (prev_char, width, height))
-        if not unicodedata.combining(line[-1]):
-            print("%s 0 0 %d %d 0" % (line[-1], width, height))
-        print("\t 0 0 %d %d 0" % (width, height))
+    if len(lines) == 1:
+        raise ValueError("ERROR: %s: Ground truth text file should contain exactly one line, not %s" % (args.txt, len(lines)))
+    line = unicodedata.normalize('NFC', lines[0].strip())
```

Comment:
Wow. This is actually a pretty strong modification, right? Are we sure that NFC is the right decision for all the exotic languages for which models are trained for using tesstrain?

Reply:
I didn't introduce the NFC normalization, I just moved the normalization step.

Reply:
Was this value error supposed to be raised for lines greater than one? because the current if condition is checking for length to be 1. I was training on my data whose line length is 1 but getting this issue:

Traceback (most recent call last):
  File "generate_line_box.py", line 32, in <module>
    raise ValueError("ERROR: %s: Ground truth text file should contain exactly one line, not %s" % (args.txt, len(lines)))
ValueError: ERROR: data/my-handwriting-ground-truth/1.gt.txt: Ground truth text file should contain exactly one line, not 1

Reply:
> Was this value error supposed to be raised for lines greater than one?

Yes. The code here used a wrong check. That is fixed in the latest code.